### DISCUSSION TITLE
Use getUsername() or getUserIdentifier() ?

### DISCUSSION THREAD

**Discussion by karaka200** [symfony/symfony#47558] [Q&A] [answered]
Which one should be used?
```
    public function getUsername(): ?string
    {
        return $this->username;
    }
```

```
    public function getUserIdentifier(): string
    {
        return (string) $this->username;
    }
```

Here @wouterj say that use `UserIdentifier` but some people say that `UserIdentifier` was deprecated:
https://symfony.com/blog/new-in-symfony-5-3-improvements-for-security-users#renamed-username-to-identifier

> Another source of confusion related to users is the concept of "username" which is used in the Symfony security. In many applications this username is not a traditional username, but an email or even some API token.
> 
> That's why in Symfony 5.3 we've decided to avoid this confusion and we've renamed "username" to "user identifier". This might require some changes in your application code (in 5.3 the old names still work but they are deprecated and in Symfony 6.0 they will be removed):
> 
> * UserInterface::getUsername() is now UserInterface::getUserIdentifier()
> * loadUserByUsername() is now loadUserByUserIdentifier(), both in user loaders and user providers
> * UsernameNotFoundException is now UserNotFoundException

**Comment by stof** [comment, marked as answer]
Where are you seeing that `UserIdentifier` is deprecated ? The blog post you quote is explicitly saying the opposite. `getUsername` is deprecated, being replaced by `getUserIdentifier`

**Reply by karaka200** [reply]
https://symfony-devs.slack.com/archives/C3EQ7S3MJ/p1662977177878279

Here I ask and two people say this.

I use Symfony 6.1, so you sy that I use `UserIdentifier` ?

**Reply by stof** [reply]
Yes, `UserIdentifier` is the one you need. `username` was the _old_ way in earlier versions of Symfony.

**Reply by karaka200** [reply]
@stof  So can I remove this from my entity:

```
    public function getUsername(): ?string
    {
        return $this->username;
    }
```

**Reply by stof** [reply]
In symfony 6.x, you can. In Symfony 5.4, you cannot (as the old name is not gone in 5.4)
